### ISSUE TITLE
Need ability to configure default private registry via command-line flag

### ISSUE THREAD

**Issue by brandond** [rancher/rke2#271]
We need to give users a way to configure a default registry for core images and charts. rke2-charts support this via a value: `global.systemDefaultRegistry`. For system images (rke2 apiserver/controller-manager/etc) and bundled controllers (helm-controller) we will need to add the ability to pass a value that is prepended to the image name, same as how the helm charts handle it.

Design goals:
* Allow users to bootstrap a system from a local private registry **without providing any local content via airgap tarball**
* Prefix core images
* Pass through to `global.systemDefaultRegistry` for core charts
* Private registry must use TLS (no plaintext HTTP), with the cert trusted by the host OS CA bundle
* Private registry must not require authentication
* Prefix must be specified as host:port

Non-goals:
* Prefix any images, or configure helm charts, other than those required to support the core system and packaged components
* Configure the default registry as a docker.io registry mirror in containerd
* Advanced registry use cases (plaintext/auth/cert validation disabled)
* Tarball image preload is not supported when --system-default-registry is configured. 
  _k3s code will still load the images, but they won't be used because the images will be tagged as coming from the wrong registry._

Stretch goals:
* Provide a way for users to take advantage of the same behavior for their own manifests

**Comment by rancher-max** [comment]
This is mostly working. The only image that isn't using the private registry is `rke2-ingress-nginx-controller`, which is therefore failing in an airgap setup. See below the comparison of the deployments from metrics-server and this image.
```
$ kubectl get -n kube-system deploy/rke2-metrics-server -o yaml | grep image
                f:image: {}
                f:imagePullPolicy: {}
        image: maxsprivateregistry.com/rancher/hardened-k8s-metrics-server:v0.3.6
        imagePullPolicy: IfNotPresent

$ kubectl get -n kube-system deploy/rke2-ingress-nginx-controller -o yaml | grep image
                f:image: {}
                f:imagePullPolicy: {}
        image: rancher/nginx-ingress-controller:nginx-0.30.0-rancher1
        imagePullPolicy: IfNotPresent
```

**Comment by brandond** [comment]
IMO @rancher-max this should not be reopened, since the issue is with the chart, not rke2 core. I'm going to close this out as working, and you can test the chart fix in #369
